### PR TITLE
Add parameter to pipeline for running BrowserStack tests

### PR DESCRIPTION
**Pull request by carzh**
### Description
Add parameter to select whether to run BrowserStack or App Center tests.

Looks like this in the UI:
image

### Motivation and Context
We are transitioning our E2E mobile tests from running on App Center to running on BrowserStack; however, we are running into some blocking account issues with BrowserStack. This change allows us to toggle between the E2E test platform as account issues are resolved.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml [hunk 1]
```diff
@@ -82,13 +87,31 @@ jobs:
...
     timeoutInMinutes: 15
-    enabled: false
     env:
       BROWSERSTACK_ID: $(browserstack_username)
       BROWSERSTACK_TOKEN: $(browserstack_access_key)
 
+  - ${{ if eq(parameters.runBrowserStackTests, false) }}:
+    - template: install-appcenter.yml
+
+  - script: |
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml:100]: Should you add one level indentation here? If appcenter was not installed, it should not run appcenter tests.

Comment (carzh) [reply tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml:100]: oh true -- the script step has a condition that checks the runBrowserStackTests parameter, but will be more readable and achieves the same goal if script step is indented

File: tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml [hunk 2]
```diff
@@ -82,13 +87,30 @@ jobs:
...
     displayName: Run E2E tests using Browserstack
+    condition: eq('${{ parameters.runBrowserStackTests }}', 'true')
     workingDirectory: $(Build.BinariesDirectory)/android_test/android/app/build/outputs/apk
     timeoutInMinutes: 15
-    enabled: false
     env:
       BROWSERSTACK_ID: $(browserstack_username)
       BROWSERSTACK_TOKEN: $(browserstack_access_key)
 
+  - ${{ if eq(parameters.runBrowserStackTests, false) }}:
```
Comment (skottmckay) [inline tools/ci_build/github/azure-pipelines/templates/android-java-api-aar-test.yml:97]: Could we use something like this to be slightly more consistent?

```yaml
- ${{ if eq(parameters.runBrowserStackTests, true) }}:
  -  <steps to run browser stack based tests>
- ${{ else }}:
  -  <steps to run app center based tests>
```

**Comment by skottmckay** [comment]
Out of interest does the option in the UI show as being ticked when the default is 'true'?

**Comment by carzh** [comment]
> Out of interest does the option in the UI show as being ticked when the default is 'true'?

yep it'll show as ticked

Closing as the other PR to add browserstack tests back in was approved and merged

**Pull request closed without merging**